src/manager/lotus.rs

```diff
@@ -306,6 +306,21 @@ impl<T: JsonRpcClient + Send + Sync> SubnetManager for LotusSubnetManager<T> {
         Address::from_str(&addr_str).map_err(|_| anyhow!("cannot get address from string output"))
     }
 
+    async fn wallet_list(&self) -> Result<Vec<Address>> {
+        log::info!("list wallet in subnet");
+        self.lotus_client
+            .wallet_list()
+            .await?
+            .iter()
+            .map(|raw| Address::from_str(raw).map_err(|_| anyhow!("invalid addr: {raw:}")))
+            .collect::<Result<_>>()
+    }
+
+    async fn wallet_balance(&self, address: &Address) -> Result<TokenAmount> {
```

Comment:
IIRC, `Filecoin.WalletList` already returns the balance for the list of addresses, I think is good to keep this `wallet_balance` RPC method here as we may need it in the future, but maybe you can save a few calls by getting all the info you need from wallet list.